### DISCUSSION TITLE
teslamate abrp integration request

### DISCUSSION THREAD

**Discussion by pleasanthill17** [teslamate-org/teslamate#2072] [Ideas]
Hi,

There are two teslamate abrp projects: https://github.com/pyjamasam/teslamate-abrp and the original project https://github.com/letienne/teslamate-abrp (Thanks to both the authors).

I wrote to ABRP to get telemetry-only api key for use to use one of the above projects, so I don't have to use tesla api with a third party.

ABRP responded to me with the following request and agreed to send me a key (not received one yet).

 "We now have a bunch of individual users with individual API keys for TeslaMate. Could you please create a PR for the TeslaMate ABRP integration and propose a common ABRP API key for all users for TeslaMate? "

I am not an expert. Infact, even after they send me the API, I am going to struggle to use above two projects. I do not understand what they are requesting and if it is even possible with teslamate. Was wondering if any of the experts in this group could shed light on this request and its feasibility. TIA !!

I have followed teslamate wiki and tutorials and it works great including home assistant integration. Thanks for such a great project !!

**Comment by fetzu** [comment]
Hey @pleasanthill17,

What they are essentially asking is that “we” coordinate and use a single API key for TeslaMate-ABRP request(s). We can try to coordinate this (see https://github.com/adriankumpf/teslamate/discussions/1592#discussioncomment-1641322)
